api/handlers.go

```diff
@@ -28,50 +44,64 @@ func (app *Application) AllArticle(w http.ResponseWriter, r *http.Request) {
 	// Retrieve the list of articles from the database
 	articles, _ := app.DB.AllArticles()
 
-	// Prepare the response map
-	response := map[string]interface{}{
-		"status":  200,
-		"message": "Success",
-		"data":    []map[string]interface{}{},
-	}
+	// Create the response struct
+	var response models.Response
+
+	response.Status = http.StatusOK
+	response.Message = "Success"
 
-	// Check if there are articles, then add them to the response map
-	if len(articles) > 0 {
-		for _, article := range articles {
+	var dataList []models.Articles // Slice to hold the articles
 
-			response["data"] = append(response["data"].([]map[string]interface{}), map[string]interface{}{"id": article.ID, "title": article.Title, "content": article.Content, "author": article.Author})
+	// Check if there are articles, then add them to the response
+	for _, article := range articles {
+		articleData := models.Articles{
+			ID:      article.ID,
+			Title:   article.Title,
+			Content: article.Content,
+			Author:  article.Author,
 		}
+		dataList = append(dataList, articleData)
 	}
 
+	// Set the response data as a slice of articles
+	response.Data = dataList
+	if len(dataList) <= 1 {
+		log.Println("No article found for the given ID")
+		utility.WriteJSON(w, http.StatusNotFound, models.Response{Data: nil, Status: http.StatusNotFound, Message: "No article found"})
+		return
+	}
 	// Set the response headers and write the JSON response
 	utility.WriteJSON(w, http.StatusOK, response)
 }
 
 func (app *Application) GetArticle(w http.ResponseWriter, r *http.Request) {
 	// Get the article ID from the URL parameter
 	id := chi.URLParam(r, "id")
-	articleID, err := strconv.Atoi(id)
-	if err != nil {
-		utility.ErrorJSON(w, err)
-		return
-	}
+	articleID, _ := strconv.Atoi(id)
```

Comment:
error handing missing should handle convert string to int error. If the user pass invalid id like string then return proper error with 400 status code